src/main/resources/com/microsoft/jenkins/containeragents/aci/AciContainerTemplate/config.jelly

```diff
@@ -49,6 +49,15 @@
                                     deleteCaption="${%Delete_Volume}"/>
     </f:entry>
 
+    <f:optionalBlock name="privateIpAddress" title="${%Use_Private_IP}" field="privateIpAddress" checked="${!empty instance.privateIpAddress}">
```

Comment:
I think it should be an `OptionalProperty` see https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/PLUGINS.md#step-4

You've gone most of the way by already creating a describeable object, you shouldn't need this `checked` field

Normally localised fields would use the english text (old ones here don't), i.e.
```suggestion
    <f:optionalBlock name="privateIpAddress" title="${%Use Private IP Address}" field="privateIpAddress" checked="${!empty instance.privateIpAddress}">
```

possibly `Enable Virtual Network integration` might be better text too.

Reply:
Thank you for your advice. I had some trouble with jelly. It was for me like try-and-error-and-googeling to write this piece of code. I will check your advice to `OptionalProperty`.

I don't know whether "Enable Virtual Network integration" is a good name because you can have a public ip and a virtual network. In this case, virtual network is not mandatory. In case of private IP address, virtual network is mandatory. IMO Introducing virtual network in common would be a new issue.